### PR TITLE
[TIMOB-24702] Android: Fix onDestroy dispatchCallback

### PR DESCRIPTION
**Pull request by garymathews**
- Fix `TiBaseActivity` `dispatchCallback`
- Fix `TiViewProxy` `MSG_GETSIZE` and `MSG_GETRECT`

```JS
var win = Titanium.UI.createWindow(),
    btn = Ti.UI.createButton({
        title: 'CLOSE'
    });

btn.addEventListener('click', function(e) {
	var activity = Ti.Android.currentActivity;
	if (!activity) {
		return;
	}

	activity.onDestroy = function(e) {
		var size = btn.size;
		var rect = btn.rect;
	};
	activity.finish();
});

win.add(btn);
win.open();
```

[JIRA Ticket](https://jira.appcelerator.org/browse/TIMOB-24702)

### CODE REVIEW

**Comment by jquick-axway** [comment]
@garymathews, you need to check if the getAppRootOrCurrentActivity() or getAppCurrentActivity() methods return null in the handleMessage() method. The reason is because an Android Handler's dispatched/posted message/runnable can still be invoked on the main UI thread even if there is no UI/activity being displayed... and you can't assume a handler's runnable/message has been invoked on-time.

That is, a Handler is not bound to an active activity. You can post a message/runnable via a handler to be invoked say 1 minute later, back out of all activities, and that posted message/runnable will still be received/invoked while the app is in the background without any UI. This is normal behavior on Android.

File: android/titanium/src/java/org/appcelerator/titanium/proxy/TiViewProxy.java [hunk 1]
```diff
@@ -296,7 +296,7 @@ public boolean handleMessage(Message msg)
 						TiDimension nativeHeight = new TiDimension(v.getHeight(), TiDimension.TYPE_HEIGHT);
 
 						// TiDimension needs a view to grab the window manager, so we'll just use the decorview of the current window
-						View decorView = TiApplication.getAppCurrentActivity().getWindow().getDecorView();
+						View decorView = TiApplication.getAppRootOrCurrentActivity().getWindow().getDecorView();
```
Comment (ypbnv) [inline android/titanium/src/java/org/appcelerator/titanium/proxy/TiViewProxy.java]: Guard for null result from getAppRootOrCurrentActivity()

File: android/titanium/src/java/org/appcelerator/titanium/proxy/TiViewProxy.java [hunk 2]
```diff
@@ -326,7 +326,7 @@ public boolean handleMessage(Message msg)
 						TiDimension nativeTop = new TiDimension(position[1], TiDimension.TYPE_TOP);
 
 						// TiDimension needs a view to grab the window manager, so we'll just use the decorview of the current window
-						View decorView = TiApplication.getAppCurrentActivity().getWindow().getDecorView();
+						View decorView = TiApplication.getAppRootOrCurrentActivity().getWindow().getDecorView();
```
Comment (ypbnv) [inline android/titanium/src/java/org/appcelerator/titanium/proxy/TiViewProxy.java]: Guard for null result from getAppRootOrCurrentActivity()

**Review by ypbnv — CHANGES_REQUESTED** [review]

**Comment by garymathews** [comment]
@ypbnv Updated PR

**Review by ypbnv — APPROVED** [review]
CR: Pass

**Review by ssjsamir — APPROVED** [review]
**FR Passed** application no longer crashes when accessing 'size' & 'rect' button properties after activity.finish().

**Test Steps**
+ Downloaded the SDK Build from this PR
+ Created a new Titanium project
+ Copied the the test case from description
+ Ran the application 
+ Pressed `Close`
+ Window closed without any issues

**Test Environment**
Appcelerator Command-Line Interface, version 6.2.2
Google Nexus 6P (7.1.1)
Operating System Name: Mac OS X El Capitan
Operating System Version: 10.11.6
Node.js Version: 6.10.1
Xcode: 8.2
Appcelerator Studio: 4.9.0.201705251638

**Pull request merged**